accounts-db/src/accounts_db.rs

```diff
@@ -4353,6 +4403,37 @@ impl AccountsDb {
         );
     }
 
+    pub(crate) fn zero_lamport_single_ref_found(&self, slot: Slot, offset: usize) {
+        if let Some(store) = self.storage.get_slot_storage_entry(slot) {
+            let mut zero_lamport_single_ref_offsets =
```

Comment:
Can we move this logic one level up, i.e. after 'index.scan(...)`?

In index.scan(...), we keep a counter of how many single_ref_zeros.
After the index.scan(), we do the check below.

In this way, we don't need zero_lamport_single_ref_offsets: RwLock<IntSet<usize>> on account_storage at all?

Can we just keep the count, zero_lamport_single_ref_counts on account_storage instead, for the later usage in  do_shrink_store()?